src/client/src/FrameOrderCoordinator.c

```diff
@@ -130,12 +130,12 @@ STATUS audioVideoFrameTimestampComparator(PFrameOrderTrackData pFrameOrderTrackD
     // if two frames from different tracks have same mkv timestamps, and the latter frame has a KEY_FRAME_FLAG set,
     // then move KEY_FRAME_FLAG to the earlier frame. Because we cant have frame before key frame that has the same
     // mkv timestamp as the key frame.
-    if (mkvTimestamp1 == mkvTimestamp2 && pFrameOrderTrackData1->pTrackInfo->trackType != pFrameOrderTrackData2->pTrackInfo->trackType) {
-        if (!firstFrameFirst) {
-            pSwap = pFirst;
-            pFirst = pSecond;
-            pSecond = pSwap;
-        }
+    if (mkvTimestamp1 == mkvTimestamp2 &&
+        pFrameOrderTrackData1->pTrackInfo->trackType != pFrameOrderTrackData2->pTrackInfo->trackType &&
+        (pSecond->flags & FRAME_FLAG_KEY_FRAME) != FRAME_FLAG_NONE) {
```

Comment:
Please use macro for this:  `CHECK_FRAME_FLAG_KEY_FRAME`

Since this is a bug fix can we get a simple unit test that hits the case that fails without the change?

Reply:
Let me add the change with the macro. I will craft a unit test as well